python/dialects/aiex.py

```diff
@@ -836,3 +836,64 @@ def dma_start_bd_chain_for(symbol, args, alloc, *pyargs, **kwargs):
     return DMAStartBdChainForOp(
         T.index(), chain_sym, args, alloc_sym, *pyargs, **kwargs
     )
+
+
+def shim_dma_bd(
+    mem,
+    offset: int | None = None,
+    sizes: MixedValues | None = None,
+    strides: MixedValues | None = None,
+    transfer_len: int | None = None,
+):
+    if offset is None:
+        offset = 0
+    if sizes is None:
+        sizes = [0] * 4
+    if strides is None:
+        strides = [0] * 3 + [1]
+
+    if transfer_len is None:
+        if len(sizes) >= 4:
+            # For shim dma bd, highest dimension is repeat count which is not included in the length
```

Comment:
I agree that the interface could use some cleaning up. For dma_memcpy_nd it made some sense to include it with the dimensions because there only ever was one BD in a task, so it behaved consistently. However, even for that, doing it that way restricts users to only be able to use the second example from above, 1 and 3 would not be possible since `dma_memcpy_nd` forces the iteration step wrap to be equal to the repeat count. For chained BDs the simplification breaks down completely i.m.o.